### PR TITLE
update string function

### PR DESCRIPTION
**Pull request by siddontang**
+ support lower/upper 
+ update length

### CODE REVIEW

File: expression/builtin/string.go [hunk 1]
```diff
@@ -116,3 +118,31 @@ func builtinRepeat(args []interface{}, ctx map[interface{}]interface{}) (v inter
 	}
 	return strings.Repeat(ch, num), nil
 }
+
+// See: https://dev.mysql.com/doc/refman/5.0/en/string-functions.html#function_lower
```
Comment (qiuyesuifeng) [inline expression/builtin/string.go]: Use 5.7 version doc ref?

File: expression/builtin/string_test.go [hunk 2]
```diff
@@ -122,3 +137,32 @@ func (s *testBuiltinSuite) TestRepeat(c *C) {
...
+
+	tbl := []struct {
+		Input  interface{}
+		Expect string
+	}{
+		{"abc", "abc"},
+		{1, "1"},
+	}
+
+	for _, t := range tbl {
```
Comment (shenli) [inline expression/builtin/string_test.go]: lower and upper use the same tbl?
upper("abc") should be "ABC"
lower("abd") should be "abc"

Comment (siddontang) [inline expression/builtin/string_test.go]: yes, you can see the following test that I use strings.ToUppoer for upper test.

Comment (shenli) [inline expression/builtin/string_test.go]: OK. I did not notice it.

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**